subprojects/docs/src/docs/userguide/java_platform_plugin.adoc

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+:maven-bom: https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Management
+:spring-boot-bom: https://docs.spring.io/spring-boot/docs/current/reference/html/using-boot-build-systems.html#using-boot-dependency-management
+:metadata-file-spec: https://github.com/gradle/gradle/blob/master/subprojects/docs/src/docs/design/gradle-module-metadata-specification.md
+
+[[java_platform_plugin]]
+= The Java Platform Plugin
+
+The Java Platform plugin brings the ability to declare platforms for the Java ecosystem.
+A platform can be used for different purposes:
+
+- a description of modules which are published together (and for example, share the same version)
+- a set of recommended versions for heterogeneous libraries. A typical example includes the {spring-boot-bom}[Spring Boot BOM]
+- <<sec:java_platform_consumption,sharing a set of dependency versions>> between subprojects
+
+A platform is a special kind of software component which doesn't contain any sources:
+it is only used to reference other libraries, so that they play well together during dependency resolution.
+
+Platforms can be published as {maven-bom}[Maven BOMs] or with the experimental {metadata-file-spec}[Gradle metadata file] format.
+
+
+[[sec:java_platform_usage]]
+== Usage
+
+To use the Java Platform plugin, include the following in your build script:
+
+.Using the Java Platform plugin
+====
+include::sample[dir="java-platform/quickstart/groovy",files="build.gradle[tags=use-plugin]"]
+include::sample[dir="java-platform/quickstart/kotlin",files="build.gradle.kts[tags=use-plugin]"]
+====
+
+
+[[sec:java_platform_separation]]
+== API and runtime separation
+
+A major difference between a Maven BOM and a Java platform is that Gradle makes the difference between dependencies required at compile time and those only needed at run time.
```

Comment:
We have sometimes "run time" and other times "runtime". I believe in software we can use "runtime" always, but happy to be corrected if wrong.

Reply:
I use it slightly differently depending on the context:

- at "run time": when the application runs
- in "runtime": the context of execution